examples/mir_demo_server/server_example.cpp

```diff
@@ -143,14 +145,22 @@ try
     InputFilters input_filters;
     me::TestClientRunner test_runner;
 
+    miral::WaylandExtensions extensions{
+        miral::WaylandExtensions::recommended_extensions()
+            + ":" + mir::examples::server_decoration_extension.name
+            + ":" + mir::examples::xdg_decoration_extension.name};
+    extensions.add_extension(mir::examples::server_decoration_extension);
+    extensions.add_extension(mir::examples::xdg_decoration_extension);
+    extensions.set_filter([&](auto, char const* protocol) { puts(protocol); return true; });
```

Comment:
I *still* don't like manually constructing a colon separated string, this be better:
```
miral::WaylandExtensions extensions;
extensions.add(mir::examples::server_decoration_extension);
extensions.add(mir::examples::xdg_decoration_extension);
extensions.set_filter([&](auto, char const* protocol) { puts(protocol); return true; });
```
(with `extensions.enable(name)` and `extensions.disable(name)` if you wanted to turn on or off default or added ones)

Reply:
I'm not sure if I prefer @wmww's suggestion here, but as an API edge I think “pass in a colon-delimited string” is something we could do better than.

For my oar, maybe something like:
```
miral::WaylandExtensions extensions{
    miral::WaylandExtensions::recommended_extensions() +
    mir::examples::server_decoration_extension +
    miral::WaylandExtensions::Disabled(mir::examples::xdg_decoration_extension)};
```

Reply:
What @wmww suggests is consistent with the approach we chose in an earlier PR (#775). This code predates that and probably shouldn't be supported.

In any case, this is isn't what I plan to land from this prototype, just experimenting with how it might be used.